test/test_sync.cpp

```diff
@@ -6942,6 +6942,338 @@ TEST(Sync_SetAndGetEmptyReciprocalChangeset)
     CHECK(reciprocal_changeset.empty());
 }
 
+TEST(Sync_SetEmptyReciprocalChangesetAfterNonEmptyReciprocalChangeset)
+{
+    using namespace realm;
+    using namespace realm::sync::instr;
+    using realm::sync::Changeset;
+
+    TEST_CLIENT_DB(db);
+
+    auto& history = get_history(db);
+    history.set_client_file_ident(SaltedFileIdent{1, 0x1234567812345678}, false);
+    timestamp_type timestamp{1};
+    history.set_local_origin_timestamp_source([&] {
+        return ++timestamp;
+    });
+
+    auto latest_local_version = [&] {
```

Comment:
it looks like this gets used by all three of these tests, maybe move it into its own static function?

Reply:
it's not the exact setup for all three tests, but I can look into refactoring it (or perhaps split it into two part: schema creation and adding the initial data)

Reply:
Ah, got it. I will leave whether to refactor this to your judgement.

Reply:
decided it's not worth it. I'd've normally implemented this using sections. maybe we should switch to using catch as well.